### PR TITLE
Include team to the store

### PR DESCRIPTION
**Pull request by enahum**
This includes Team to the store and Actions against the platform which handles most of the API's for teams

### CODE REVIEW

File: src/reducers/entities/teams.js [hunk 1]
```diff
@@ -16,60 +17,127 @@ function currentId(state = '', action) {
...
 function membersInTeam(state = {}, action) {
     switch (action.type) {
-    default:
-        return state;
+    case TeamsTypes.RECEIVED_MEMBERS_IN_TEAM: {
+        const data = action.data;
+        let teamId;
+        let members;
+
+        if (Array.isArray(data)) {
```
Comment (hmhealey) [inline src/reducers/entities/teams.js]: Shouldn't `action.data` always be a consistent type? It seems weird for it to switch

File: src/reducers/entities/teams.js [hunk 2]
```diff
@@ -16,60 +17,127 @@ function currentId(state = '', action) {
...
+            teamId = data[0].team_id;
+            members = new Set(state[teamId]);
```
Comment (hmhealey) [inline src/reducers/entities/teams.js]: This line is the same in both branches of the if statement

File: src/reducers/entities/teams.js [hunk 3]
```diff
@@ -16,60 +17,127 @@ function currentId(state = '', action) {
 function teams(state = {}, action) {
     switch (action.type) {
     case TeamsTypes.RECEIVED_ALL_TEAMS:
+    case TeamsTypes.RECEIVED_TEAM_LISTINGS:
         return Object.assign({}, state, action.data);
+
+    case TeamsTypes.CREATED_TEAM:
+    case TeamsTypes.UPDATED_TEAM:
+        return {...state, [action.data.id]: action.data};
```
Comment (hmhealey) [inline src/reducers/entities/teams.js]: Do you mind splitting this onto multiple lines to keep it consistent?

File: src/reducers/entities/teams.js (hunk 1 above)
Comment (enahum) [inline src/reducers/entities/teams.js]: I could have different constants for this if you prefer...

File: src/reducers/entities/teams.js (hunk 2 above)
Comment (enahum) [inline src/reducers/entities/teams.js]: but they depend on the teamId which is dependent of the if array, but nvm I'm going to have different constants for this one

**Comment by enahum** [comment]
@hmhealey done

**Pull request merged**